lib/java/com/google/android/material/bottomsheet/BottomSheetDialogFragment.java

```diff
@@ -31,4 +38,38 @@
   public Dialog onCreateDialog(Bundle savedInstanceState) {
     return new BottomSheetDialog(getContext(), getTheme());
   }
+
+  @Override
+  public void dismiss() {
```

Comment:
I think both `dismissAllowingStateLoss()` and `dismiss()` will need to have the same logic which prevents the fragment from being dismissed immediately.

Could this method do something similar to cancel in `BottomSheetDialog`? If the bottom sheet is hidden, call through to super. Otherwise just set the state of the BottomSheet to hidden. That should call through to cancel the dialog, which should call `onCancel(DialogInterface dialog)` in `BottomSheetDialogFragment` which can be modified to call `dismiss()`

Reply:
Yep, I can do that 👍

Reply:
Done 👍